src/Symfony/Component/Form/Extension/DependencyInjection/DependencyInjectionExtension.php

```diff
@@ -82,8 +114,13 @@ public function getTypeGuesser()
             $this->guesserLoaded = true;
             $guessers = array();
 
-            foreach ($this->guesserServiceIds as $serviceId) {
-                $guessers[] = $this->container->get($serviceId);
+            foreach ($this->guesserServices as $serviceId => $service) {
+                if (is_string($service)) {
+                    $serviceId = $service;
+                    $service = $this->typeContainer->get($serviceId);
+                }
+
+                $guessers[] = $service;
```

Comment:
I thought about deprecating the whole class and replacing it with a new one. But after discussing with @nicolas-grekas it requires too much efforts then to maintain both implementations. This is why we choose to adapt the class itself. One class to maintain and one tests suite. It's just a bit more difficult to read and understand the code at the moment because of the mixed two layers.